src/libraries/System.Private.CoreLib/src/System/Diagnostics/StackTrace.cs

```diff
@@ -322,6 +324,21 @@ internal void ToString(TraceFormat traceFormat, StringBuilder sb)
                             sb.Append(' ');
                             sb.AppendFormat(CultureInfo.InvariantCulture, inFileLineNum, fileName, sf.GetFileLineNumber());
                         }
+                        else if (s_ilOffsetToStackTrace)
+                        {
+                            try
+                            {
+                                string methodToken = mb.MetadataToken.ToString("x", CultureInfo.InvariantCulture);
+                                string ilOffset = sf.GetILOffset().ToString("x", CultureInfo.InvariantCulture);
```

Comment:
Actually, this could be better done with `HexConverter`

Reply:
Thank you. It seems better to use Convert like below.
```
    string methodToken = Convert.ToString(mb.MetadataToken, 16);
    string ilOffset = Convert.ToString(sf.GetILOffset(), 16);
````

Reply:
Please avoid Convert too, it's unnecessary complex. The code could be as simple as

```c#
Span<byte> bytes = stackalloc byte[4];
BinaryPrimitives.WriteUInt32BigEndian(bytes, mb.MetadataToken);
string methodToken = HexConverter.ToString(bytes.Slice(0, 4), HexConverter.Casing.Lower);
```

Reply:
I'll modify it to use HexConverter like your suggestion.